textarea/textarea.go

```diff
@@ -0,0 +1,1246 @@
+package textarea
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+	"unicode"
+
+	"github.com/atotto/clipboard"
+	"github.com/charmbracelet/bubbles/viewport"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	rw "github.com/mattn/go-runewidth"
+)
+
+const defaultBlinkSpeed = time.Millisecond * 530
```

Comment:
I almost wonder if we should, somehow, move the cursor into its own bubble?

It seems like it would be the same thing for both textarea and textinput